hoodie-client/src/main/java/com/uber/hoodie/config/HoodieMemoryConfig.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.config;
+
+import javax.annotation.concurrent.Immutable;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * Memory related config
+ */
+@Immutable
+public class HoodieMemoryConfig extends DefaultHoodieConfig {
+
+  // This fraction is multiplied with the spark.memory.fraction to get a final fraction of heap space to use during merge
+  // This makes it easier to scale this value as one increases the spark.executor.memory
+  public static final String MAX_MEMORY_FRACTION_FOR_MERGE_PROP = "hoodie.merge.memory.fraction";
+  // Default max memory fraction during hash-merge, excess spills to disk
+  public static final String DEFAULT_MAX_MEMORY_FRACTION_FOR_MERGE = String.valueOf(0.6);
+  public static final String SPARK_EXECUTOR_MEMORY_PROP = "spark.executor.memory";
+  public static final String SPARK_EXECUTOR_MEMORY_FRACTION_PROP = "spark.memory.fraction";
+  public static final String DEFAULT_SPARK_EXECUTOR_MEMORY_FRACTION = "0.6";
+  public static final String DEFAULT_SPARK_EXECUTOR_MEMORY = "1024"; // in MB
+  // Use this in case SparkEnv is absent
+  public static final String DEFAULT_MAX_MEMORY_IN_BYTES_FOR_MERGE = String
+      .valueOf(1024 * 1024 * 1024L); // 1GB
+
+  // TODO(NA):convert this to fraction, use either Runtime.freeMemory() or spark.executor.memory fractions depending on execution
+  public static final String MAX_SIZE_IN_MEMORY_PER_COMPACTION_IN_BYTES_PROP = "hoodie.compaction.spill.threshold";
```

Comment:
I feel this should stay in `CompactionConfig` or be renamed.. High level, its debatable if we need a class for these two configs.. I am with you on bringing all the memory management related stuff into a place, but this list feels too small..  thoughts?

Reply:
I think separation makes sense right now.

Reply:
lets chat more.. Not in favor of introducing new classes, while deferring their real usage down the line.. Happy to consider this if we can group more memory related params.. are there others that can live here?

Reply:
This is not exactly deferring their usage down the line, there's more than 1 memory config being used here and adding them to the HoodieWriteConfig or HoodieCompactionConfig seems like a wrong place. But if you feel strongly about not adding another class, I can move these configs to HoodieWriteConfig.

Reply:
I dont think we are going to agree. I ll bite.. but can we rename `hoodie.compaction.spill.threshold` to `hoodie.memory.compaction.spill.threshold` to be consistent?

Reply:
Renamed it!